### PR TITLE
OMF Temperature Sensor Sample

### PR DESCRIPTION
**Pull request by dcbroad3**

### CODE REVIEW

File: advanced_samples/TemperatureSensor/Python/azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    steps:
+      - template: '/miscellaneous/build_templates/config.yml'
+        parameters:
+          fileOriginal: $(CredFile)
+          fileLibrary: $(CredFileLibraryOCS)
+          filePath: $(projPath)
+
+      - task: UsePythonVersion@0
+        inputs:
+          versionSpec: '3.6'
```
Comment (derekendres) [inline advanced_samples/TemperatureSensor/Python/azure-pipelines.yml:54]: any reason for 3.6 not latest of 3.8? https://www.python.org/downloads/ ?

File: advanced_samples/TemperatureSensor/Python/program.py [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+USE_COMPRESSION = False
+
+# OMF Destinations
+OcsOmfUrl = None
+EdsOmfUrl = None
+PiOmfUrl = None
+
+# Set this to True if your cert is trusted by the Python certify.
+# Set to False if you do not want to check the certificate (NOT RECOMMENDED)
+VERIFY_SSL = False
```
Comment (derekendres) [inline advanced_samples/TemperatureSensor/Python/program.py:35]: Default to True then?

File: advanced_samples/TemperatureSensor/Python/program.py [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+            data=msg_body,
+            verify=VERIFY_SSL,
+            timeout=WEB_REQUEST_TIMEOUT_SECONDS,
+            auth=(userName, password))
+        checkResponse(response, msg_headers, message_type, message_omf_json)
+
+
+def checkResponse(response, msg_headers, message_type, message_omf_json):
+    # Send the request, and collect the response
+    if response.status_code == 409:
```
Comment (derekendres) [inline advanced_samples/TemperatureSensor/Python/program.py:111]: Why are not worried about this error?  
Conflict code right?  So maybe just a comment about why we aren't worried about that

File: advanced_samples/TemperatureSensor/Python/program.py [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+    }
+
+    if(USE_COMPRESSION):
+        msg_headers["compression"] = compression
+
+    return msg_headers
+
+
+def getToken():
+    # Gets the oken for the omfsendpoint
```
Comment (derekendres) [inline advanced_samples/TemperatureSensor/Python/program.py:174]: oken to token

File: advanced_samples/TemperatureSensor/Python/program.py [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+    global omfVersion
+
+    action = 'create'
+    oneTimeSendType(containerId, action)
+    oneTimeSendContainer(containerId, action)
+    oneTimeSendData(containerId, action)
+
+
+def oneTimeSendDeletes(containerId):
+    global omfVersion
```
Comment (derekendres) [inline advanced_samples/TemperatureSensor/Python/program.py:218]: needed?

File: advanced_samples/TemperatureSensor/Python/program.py [hunk 6]
```diff
@@ -0,0 +1,521 @@
...
+        oneTimeSendType(containerId, action)
+    except Exception as ex:
+        print()
+        # Ignore errors in deletes to ensure we clean up as much as possible
+        print(("Error in deletes: {error}".format(error=ex)))
+        print()
+
+
+def oneTimeSendType(containerId, action):
+    global omfVersion
```
Comment (derekendres) [inline advanced_samples/TemperatureSensor/Python/program.py:247]: Needed?

File: advanced_samples/TemperatureSensor/Python/program.py [hunk 7]
```diff
@@ -0,0 +1,521 @@
...
+                },
+                "Location": {
+                    "type": "string",
+                    "description": "Location of the asset"
+                }
+            }
+        },
+        {
+            "id": TYPE_ID,
+            "name": "Temperature float type",
```
Comment (derekendres) [inline advanced_samples/TemperatureSensor/Python/program.py:293]: Seems odd to have the id configurable, but then name is stuck.  Maybe just mentioned in documentation?

File: advanced_samples/TemperatureSensor/Python/program.py [hunk 8]
```diff
@@ -0,0 +1,521 @@
...
+        }
+    ], action)
+
+
+def oneTimeSendContainer(containerId, action):
+    # OMF Container message to create a container for our measurement
+    sendOmfMessageToEndpoint("container", [
+        {
+            "id": containerId,
+            "name": "Temperature",
```
Comment (derekendres) [inline advanced_samples/TemperatureSensor/Python/program.py:317]: Seems odd to have the id configurable, but then name is stuck.  Maybe just mentioned in documentation?

File: advanced_samples/TemperatureSensor/Python/program.py (hunk 7 above)
Comment (derekendres) [reply advanced_samples/TemperatureSensor/Python/program.py:293]: so i think maybe an added section that talks about these configuration options but also explaining that a lot of it is hard coded.  My thought is  maybe after PI WEb API endpoint.  ###Sample Configuration or something like that and a breif mention of the options and what it does (and doesn't do)

File: advanced_samples/TemperatureSensor/Python/program.py [hunk 9]
```diff
@@ -309,19 +300,19 @@ def oneTimeSendType(containerId, action):
     ], action)
 
 
-def oneTimeSendContainer(containerId, action):
+def oneTimeSendContainer(action):
     # OMF Container message to create a container for our measurement
     sendOmfMessageToEndpoint("container", [
         {
-            "id": containerId,
+            "id": "PIWorld2020.Script.SL6658.Temperature",
```
Comment (derekendres) [inline advanced_samples/TemperatureSensor/Python/program.py:307]: seems oddly hardcoded here

File: advanced_samples/TemperatureSensor/Python/azure-pipelines.yml (hunk 1 above)
Comment (dcbroad3) [reply advanced_samples/TemperatureSensor/Python/azure-pipelines.yml]: Updated to 3.8

File: advanced_samples/TemperatureSensor/Python/program.py (hunk 2 above)
Comment (dcbroad3) [reply advanced_samples/TemperatureSensor/Python/program.py]: Defaulted to True

File: advanced_samples/TemperatureSensor/Python/program.py (hunk 3 above)
Comment (dcbroad3) [reply advanced_samples/TemperatureSensor/Python/program.py]: This was in the code from Zdenek and I think it's not necessary. I removed it, if we run into failures I'll see if there's a reason we need to add it back.

File: advanced_samples/TemperatureSensor/Python/program.py (hunk 4 above)
Comment (dcbroad3) [reply advanced_samples/TemperatureSensor/Python/program.py]: Fixed (as well as omfsendpoint)

File: advanced_samples/TemperatureSensor/Python/program.py [hunk 10]
```diff
@@ -0,0 +1,521 @@
...
+    if token is None:
+        raise Exception("Failed to retrieve Token")
+
+    __expiration = float(token['expires_in']) + time.time()
+    __token = token['access_token']
+    return __token
+
+
+def oneTimeSendCreates(containerId):
+    global omfVersion
```
[the remark this answers is not in the document]
Comment (dcbroad3) [reply advanced_samples/TemperatureSensor/Python/program.py]: Removed

File: advanced_samples/TemperatureSensor/Python/program.py (hunk 5 above)
Comment (dcbroad3) [reply advanced_samples/TemperatureSensor/Python/program.py]: Removed

File: advanced_samples/TemperatureSensor/Python/program.py (hunk 6 above)
Comment (dcbroad3) [reply advanced_samples/TemperatureSensor/Python/program.py]: Removed

File: advanced_samples/TemperatureSensor/Python/program.py (hunk 7 above)
Comment (dcbroad3) [reply advanced_samples/TemperatureSensor/Python/program.py]: Switched to variables just for TYPE_ID and CONTAINER_ID (hard-coded) and added a note to the README

File: advanced_samples/TemperatureSensor/Python/program.py (hunk 8 above)
Comment (dcbroad3) [reply advanced_samples/TemperatureSensor/Python/program.py:308]: Switched to variables just for TYPE_ID and CONTAINER_ID (hard-coded) and added a note to the README

File: advanced_samples/TemperatureSensor/Python/program.py (hunk 9 above)
Comment (dcbroad3) [reply advanced_samples/TemperatureSensor/Python/program.py]: Fixed

**Review by derekendres — APPROVED** [review]

**Pull request merged**